### PR TITLE
Send notifications on multisig actions

### LINKED ISSUE #3221: Notifications: Send notifications for multi sig decision method
### Story

Example: As a *user with notifications enabled*, I want to *receive notifications relating to actions using multi sig as a decision method*, so that *I am kept informed*

### Description

Notifications should be sent for each lifecycle event of multi sig motions.

```
- Creation:
    - On creation of multi-sig action.
    - Should inform user if they are required to approve.
    - Should also trigger a mention notification to any recipients.
- Approval:
    - On approval threshold being met.
- Rejection:
    - On rejection threshold being met.
- Execution:
    - When the action has been executed.
```

Documentation here: https://www.notion.so/colony/Notifications-2c4b006528b64094b8d0741f2fc293bd?pvs=4#58cdeb11a1e04bfe81613fa39f5a3953

### This issue is complete when...

- [ ] Notifications are sent for each stage of a multi sig motion

### PR DESCRIPTION
**Pull request by davecreaser**
## Description

- Send notifications for multisig actions. This includes creation, full approval, rejection and finalization.

Part of the work for this PR was done in the block ingestor here: https://github.com/JoinColony/block-ingestor/pull/275

## Testing

Ensure that the following is added to your .env.local.secrets and run npm run prepare.

```
MAGICBELL_API_KEY=<ask me for this>
MAGICBELL_API_SECRET=<ask me for this>
MAGICBELL_DEV_KEY=
```

Also ensure the API_KEY and SECRET are both in the .env file inside `amplify/backend/function/createUserNotificationsData`

* Log in as Leela in one browser, and Amy in another incognito browser. As Leela, install the multisig extension. Set the threshold to be fixed at "2".
Screenshot 2024-10-08 at 11 43 13

* Give Leela and Amy multisig owner permissions. Check that you receive notifications for these.
Screenshot 2024-10-08 at 11 47 57
Screenshot 2024-10-08 at 11 48 52
Screenshot 2024-10-08 at 11 49 25

* Create a multisig action. It can be any type of action as long as it is multisig. Check that you receive a notification when this is created.
Screenshot 2024-10-08 at 11 50 21
Screenshot 2024-10-08 at 11 50 39

* As Amy, add a second approval to the multi sig action. Go back to Leela and check that she received a new notification for the approval. Then finalize the motion as Leela, and check that she also received a notification for that.
Screenshot 2024-10-08 at 11 51 43
Screenshot 2024-10-08 at 11 52 12
Screenshot 2024-10-08 at 11 52 41

* As Leela, make another multisig motion. Immediately go back to the approval step in the sidebar and remove her approval, then reject the motion. Check that a notification arrived for this rejection.
Screenshot 2024-10-08 at 11 53 14
Screenshot 2024-10-08 at 11 53 23
Screenshot 2024-10-08 at 11 53 51

## Diffs

**New stuff** ✨

* New component: `MultisigNotificationMessage`
* New notification types

**Changes** 🏗

* Changes to some notification categories to match specs
* New block ingestor hash, and more changes there: https://github.com/JoinColony/block-ingestor/pull/275

Resolves #3221

### CODE REVIEW

**Review by rdig — CHANGES_REQUESTED** [review]
I've tested this but everything breaks when trying to approve the multisig as Amy, but the transaction never goes through

Screenshot from 2024-10-09 11-36-01

This might actually be the issue that got solved in #3276 and you just need to rebase

Here's how far I got with testing:

Screenshot from 2024-10-09 11-28-40
Screenshot from 2024-10-09 11-29-10
Screenshot from 2024-10-09 11-29-38
Screenshot from 2024-10-09 11-29-50
Screenshot from 2024-10-09 11-30-26
Screenshot from 2024-10-09 11-30-38
Screenshot from 2024-10-09 11-33-50

**Review by bassgeta — CHANGES_REQUESTED** [review]
Code changes super straight forward and readable, I'm loving the structure of notifications so far :100: 
Alright so creating a multisig motion does indeed send `leela` notifications:
image
But my dev env is plagued by the same issue @rdig  pointed out, so I can't enable notifications as `amy` :( maybe we should consider rebasing (and hoping everything advanced payment related won't be a headache :D )

Rejection works:
image

So I think this all actually works, BUT I would like to run it through all the required paces with 2 users just for sanity's sake

**Review by bassgeta — APPROVED** [review]
Thanks for rebasing, everything works as it should, great job finding such a slick solution to a problem that was quite a pain at first glance!
both leela and amy received notifications for multisig roles updated :heavy_check_mark: 
image
Both got notifications when the multisig is created :heavy_check_mark: 
image
image
Both got notifications when its threshold passed :heavy_check_mark: 
image
image

Both get it for when the motion is finalized :heavy_check_mark: 
image
image

Both get it when it's rejected:
image
image

:ship:

File: src/components/common/Extensions/UserHub/partials/NotificationsTab/partials/Notification/Action/MultisigNotificationMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  const Message = useMemo(() => {
+    if (!creator || !action || !notificationType) {
+      return formatText(MSG.unknownAction);
+    }
+
+    if (notificationType === NotificationType.MultiSigActionCreated) {
+      return (
+        <>
+          {action.metadata?.customTitle}: {actionMetadataDescription}
```
Comment (iamsamgibbs) [inline src/components/common/Extensions/UserHub/partials/NotificationsTab/partials/Notification/Action/MultisigNotificationMessage.tsx:87]: Could we add something in here that makes it clear it is a multi-sig motion and not just a regular action? Currently the notification looks exactly the same whether it is a permissions action or a multi-sig motion (the custom title is the only difference here).

Screenshot 2024-10-10 at 14 50 42
 
Maybe just sticking `Multi-sig motion created:` at the start or something. Might need checking with product.

**Review by iamsamgibbs — APPROVED** [review]
This looks great Dave! All very sensible code changes. I've left one comment about wording which would be nice to address here.

The only other thing missing from the spec is when a multi-sig motion is created, it should inform the user if they are required to approve. It's unclear from the specs whether this should be included in the wording for the multi-sig action created notification, or as an entirely separate notification. Either way, I think this can be addressed as it's own issue as it'll be a fairly big chunk of work with either option.

Otherwise, this all works great! I receive notifications when creating a multi-sig motion, approving and finalizing:
 
Screenshot 2024-10-10 at 14 54 36

And when rejecting:

Screenshot 2024-10-10 at 14 52 49

Completely unrelated, but I noticed the action type for minting via multi-sig is missing, presumably this is an issue on master:

Screenshot 2024-10-10 at 14 50 52

Nice job!

**Pull request merged**